src/components/Screen.tsx

```diff
@@ -14,24 +14,43 @@ import {
 } from '../core';
 
 // Native components
-import ScreenNativeComponent from '../fabric/ScreenNativeComponent';
-import ModalScreenNativeComponent from '../fabric/ModalScreenNativeComponent';
+import NativeScreen, {
+  NativeProps as ScreenNativeProps,
+} from '../fabric/ScreenNativeComponent';
+import ModalScreenNative, {
+  NativeProps as ModalScreenNativeProps,
+} from '../fabric/ModalScreenNativeComponent';
 import { usePrevious } from './helpers/usePrevious';
 
-type NativeScreenProps = Omit<
-  ScreenProps,
-  'sheetInitialDetentIndex' | 'sheetLargestUndimmedDetentIndex'
-> & {
-  sheetInitialDetent: number;
-  sheetLargestUndimmedDetent: number;
-};
-
-export const NativeScreen: React.ComponentType<NativeScreenProps> =
-  ScreenNativeComponent as React.ComponentType<NativeScreenProps>;
 const AnimatedNativeScreen = Animated.createAnimatedComponent(NativeScreen);
-const AnimatedNativeModalScreen = Animated.createAnimatedComponent(
-  ModalScreenNativeComponent as React.ComponentType<NativeScreenProps>,
-);
+const AnimatedNativeModalScreen =
+  Animated.createAnimatedComponent(ModalScreenNative);
+
+const AnimatedNativeModalScreenNormalized = React.forwardRef<
+  View,
+  ScreenNativeProps | ModalScreenNativeProps
+>(function AnimatedNativeModalScreenNormalized(props, ref) {
+  const { stackAnimation } = props;
+
+  let resolvedStackAnimation: ModalScreenNativeProps['stackAnimation'];
+
+  if (
+    stackAnimation === 'ios_from_right' ||
+    stackAnimation === 'ios_from_left'
+  ) {
+    resolvedStackAnimation = 'default';
+  } else {
+    resolvedStackAnimation = stackAnimation;
+  }
+
+  return (
+    <AnimatedNativeModalScreen
+      {...props}
+      ref={ref}
+      stackAnimation={resolvedStackAnimation}
+    />
+  );
+});
```

Comment:
What is this, why do we need it?

Reply:
ScreenNative and ModalScreenNative have different animations available, I don't think it's possible to satisfy TS differently without cheating (using 'as')

Reply:
No needed anymore after sync from: https://github.com/software-mansion/react-native-screens/pull/2423/commits/4836f0b9850b9e11bb61d20fe7c8de2beec670da